asmx/src/org/objectweb/asmx/ClassReader.java

```diff
@@ -940,7 +940,7 @@ public void accept(
                         w += 2;
                         for (; k > 0; --k) {
                             w = readTypeAnnotationValues(w,
-                                  c, mv, j != 0);
+                                  c, mv, j != 0, true);
```

Comment:
How about introducing two new methods, leaving the existing one unchanged. All the calls that pass `false` remain unchanged and call the original method.
You add a method with the additional boolean flag. The existing method calls the helper with `false`.
A new method `readTypeAnnotationValuesBodies` (or some other more suitable name for this specific location) is called only here and passes `true` to the call of the helper.

Reply:
Sorry I found "leave the existing one unchanged" a bit confusing... I suppose you mean I should let `readTypeAnnotationValues` have its original interface, and move all its internal content to a new helper method with the additional boolean argument? Then `readTypeAnnotationValues` calls the helper with `false`, and another new method `readTypeAnnotationValuesBodies` which is called only here calls the helper with `true`?

Reply:
Yes, that's what I meant. Sorry for not expressing that more precisely.